detection_rules/rule.py

```diff
@@ -462,6 +460,61 @@ class ThresholdCardinality:
     threshold: ThresholdMapping
 
 
+@dataclass(frozen=True)
+class NewTermsRuleData(QueryRuleData):
+    """Specific fields for new terms field rule."""
+
+    @dataclass(frozen=True)
+    class NewTermsMapping(MarshmallowDataclassMixin):
+        @dataclass(frozen=True)
+        class HistoryWindowStart:
+            field: definitions.NonEmptyStr
+            value: definitions.NonEmptyStr
```

Comment:
Nit: It's a shame that we don't add validation to the value.

Reply:
Digging into this at the moment. In [L188](https://github.com/elastic/detection-rules/blob/c6f5d47cdfafa77505463f2263952acb20cc607a/detection_rules/rule.py#LL188) in `rule.py` we only ensure that the value of this field is a string.

In reference to [L478](https://github.com/elastic/detection-rules/blob/c6f5d47cdfafa77505463f2263952acb20cc607a/detection_rules/rule.py#L478) of `rule.py`, it seems that we have some additional logic for EQL rules because the possibility of conflicts between maxspax, lookback, and internal fields which is why they are methods of the `EQLRuleData` class and not inside `QueryRuleData`.

The `look_back` method of the `EQLRuleData` class is only called in the unit test `test_all_rules.test_eql_lookback` and converts, for example `now-9m` into milliseconds, ultimately to compare it with the maxspan value of an EQL rule. to ensure it is greater than.

So in all, it appears we have no validation of the `from` field for rules, but we have comparison logic to ensure rule devs have entered maxspan and interval ranges that would not conflict.

Reply:
Added new issue for this -> https://github.com/elastic/detection-rules/issues/2416